src/torrent.cpp

```diff
@@ -1028,11 +1031,23 @@ bool is_downloading_state(int const st)
 		if (mask & torrent_flags::stop_when_ready)
 			stop_when_ready(bool(flags & torrent_flags::stop_when_ready));
 		if (mask & torrent_flags::disable_dht)
-			m_enable_dht = !bool(flags & torrent_flags::disable_dht);
+		{
+			bool const new_value = !bool(flags & torrent_flags::disable_dht);
+			if (m_enable_dht != new_value) set_need_save_resume();
```

Comment:
Maybe compare old and new flags at once?

Reply:
what do you mean? You mean also move the assignment `m_enable_dht = new_value` into the if-block?

Reply:
No. I mean something like `if (flags != m_flags) set_need_save_resume();` once per `set_flags()` instead of checking each flag independently.

Reply:
ah, I'll take a closer look at that. I don't think all flags affect the resume data (although, most do).

Reply:
one problem with that is that the flags aren't stored as flags internally, they are individual bits in a bitfield. It would be an improvement and simplification to store them as flags internally too. I think I'll make that change in `master` though.

The reason for this is that these features were originally controlled by individual API calls (as you may recall), and using flags was a way to simplify the API, and also make it more extendable without breaking ABI. The internal representation hasn't caught up yet.